beetsplug/convert.py

```diff
@@ -183,11 +183,11 @@ def encode(self, command, source, dest, pretend=False):
             self._log.info(u'Encoding {0}', util.displayable_path(source))
 
         # Substitute $source and $dest in the argument list.
-        args = shlex.split(command)
+        args = shlex.split(command.decode(ui._arg_encoding()))
         for i, arg in enumerate(args):
             args[i] = Template(arg).safe_substitute({
-                'source': source,
-                'dest': dest,
+                'source': source.decode(ui._arg_encoding()),
+                'dest': dest.decode(ui._arg_encoding()),
             })
```

Comment:
I discovered that if you instantiate shlex, you can pass a file argument, thus in that case it could support bytes if a BytesIO or similar was passed to it.